src/sys/wasi/mod.rs

```diff
@@ -186,12 +187,14 @@ impl Selector {
         };
         if let Some(index) = subscriptions.iter().position(predicate) {
             subscriptions.swap_remove(index);
-            /* NOTE: can't have read and write subscriptions.
             // We might have two subscriptions (read and write).
-            if let Some(index) = subscriptions.iter().skip(index).position(predicate) {
-                subscriptions.swap_remove(index);
+            loop {
```

Comment:
I think entire if-else block can also be replace with `while let Some(index) = subscriptions.iter().position(predicate) { subscriptions.swap_remove(index); }` Then we just need to check if any subscription was removed to return the `NotFound` error.

Reply:
done